src/main/java/seedu/address/model/moduleclass/Time.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.moduleclass;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+public class Time {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Time should only contain numbers in the following format: HH:mm";
+    public static final String VALIDATION_REGEX = "([01]?[0-9]|2[0-3]):[0-5][0-9]";
+    private final LocalTime value;
+
+    /**
+     * Represents a moduleClass's time in the address book.
+     * Guarantees: immutable; is valid as declared in {@link #isValidTime(String)}
+     */
+    public Time(String time) {
+        requireNonNull(time);
+        checkArgument(time.matches(VALIDATION_REGEX) && isValidTime(time), MESSAGE_CONSTRAINTS);
+        this.value = LocalTime.parse(time);
+    }
+
+
+    /**
+     * Returns true if a string is valid DateTime.
+     *
+     * @param test the given String.
+     * @return True if the string can be parsed and is a valid time, false otherwise.
+     */
+    public static boolean isValidTime(String test) {
+        try {
+            LocalTime.parse(test);
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
+    }
```

Comment:
I think both the regex check and the try catch block should be here. This allows line 22 to be 
```Java
checkArgument(isValidTime(time), MESSAGE_CONSTRAINTS);
```
```suggestion
    public static boolean isValidTime(String test) {
        try {
            LocalTime.parse(test);
        } catch (DateTimeParseException e) {
            return false;
        }
        return test.matches(VALIDATION_REGEX);
    }
```